calculus/tests.js

```diff
@@ -79,4 +79,38 @@ describe('evaluate_function', function() {
             assert.equal(json.result, 1/x);
         });
     });
+});
+
+
+
+
+
+describe('integral tests', function() {
+   it('basic OK test', function() {
+       const fx = '10y+4x^2';
+       const x = 'x';
+       let res = compute_integral(fx,x);
+       assert.equal(res.result,"10 * y * x + x ^ 3 * 4 / 3");
+       assert.equal(res.error, null);
+   });
+   it('test with missing variable', function() {
+       const fx = '10y+4x^2';
+       let res = compute_integral(fx);
+       assert.equal(res.result,null);
+       assert.equal(res.error, "Unexpected type of argument in function integral (expected: string or SymbolNode or number or boolean, actual: undefined, index: 1)");
+   });
+   it('test with a variable missing in the function', function() {
+      const fx =  '10y+4x^2';
+      const z = 'z';
+      let res = compute_integral(fx,z);
+      assert.equal(res.result,"(10 y + 4 x ^ 2) * z");
+      assert.equal(res.error, null);
+   });
+   it('test with null function', function() {
+       const fx = null;
+       const x = 'x';
+       let res = compute_integral(fx,x);
+       assert.equal(res.result, null);
+       assert.equal(res.error, "Unexpected type of argument in function integral (expected: string or Node or number or boolean, actual: null, index: 0)");
```

Comment:
Same, check the start, not the whole message

Reply:
same, done ;)